ios/MagnetScannerReact.swift

```diff
@@ -14,12 +17,41 @@ import Foundation
   }
   
   @objc func stop() {
-    scanner.stop()
+    // We don't stop scanning in iOS, we let the OS to
+    // throttle the notifications of beacons found.
+    // scanner.stop()
   }
   
   func onItemFound(item: Dictionary<String, AnyObject>) {
-    print("item found", item["url"]);
+    print("item found", item["url"])
     bridge.eventDispatcher()
-      .sendDeviceEventWithName("magnetscanner:itemfound", body: item);
+      .sendDeviceEventWithName("magnetscanner:itemfound", body: item)
+    
+    notifyUser(item["url"] as! String)
+  }
+  
+  // Throttles the notification process, waiting for 1 minute until
+  // setting up the badge with the number of elements nearby.
+  func notifyUser(url: String) {
+    if (toNotify.contains(url)) {
+      return;
+    }
+    toNotify.append(url)
+    if (notifyTimer != nil) {
+      notifyTimer.invalidate();
+    }
+    
+    // Run this timer in the main loop otherwise our notifications won't show up.
+    notifyTimer = NSTimer.init(timeInterval: 60 ,
+                               target: self,
+                               selector: #selector(MagnetScannerReact.doNotifyUser),
```

Comment:
is a way of specifying a callback, before swift 2.2 those were strings and prone to commit errors, now they are not, you can check:

https://developer.apple.com/library/ios/documentation/General/Conceptual/DevPedia-CocoaCore/Selector.html